src/layout.cpp

```diff
@@ -64,9 +57,12 @@ HSFrame::~HSFrame() = default;
 
 HSFrameLeaf::HSFrameLeaf(HSTag* tag, Settings* settings, weak_ptr<HSFrameSplit> parent)
     : HSFrame(tag, settings, parent)
-    , selection(0)
 {
-    layout = (LayoutAlgorithm) settings->default_frame_layout();
+    auto l = settings->default_frame_layout();
+    if (l >= layoutAlgorithmCount()) {
```

Comment:
Isn't this check already performed by the validator of `default_frame_layout`? Do we really need it here?

Reply:
In the sense of modularity: the cast from `int` to `LayoutAlgorithm` happens here but with only 'local knowledge', we don't know that the range of the int has already been checked. This check does not cost much, and it does not make the code ugly, so I'd vote for keeping it.

Reply:
Okay, agreed.

I suppose the "real" fix would be not using an `int` internally when we already know it to be a `LayoutAlgorithm`. Would it be possible/feasible to turn the attribute into a `LayoutAlgorithm` (not right now, but in general)?

Reply:
In principle yes. The current interface for the user is quite inconvenient, because the user has to enter a number instead of a layout name.